### PR TITLE
609 ingredients needs to be visible in meal page

### LINKED ISSUE #609: [FEATURE] Ingredients needs to visible in Meal page
**Is your feature request related to a problem? Please describe.**
Ingredients need to be visible for each meal in the mealplanner UI
**Describe the solution you'd like**
 Meal page should have the ingredients listed as per the meal recipe along with quantity and unit.

**Describe alternatives you've considered** && **Additional context**
We had measures and product only earlier. The product name was same as ingredient name earlier. Now  we want to distinguish between ingredient and product. Hence we created an ingredient table and match table. So Ingredient needs to queried from the respective table  for the respective meal.

### PR DESCRIPTION
**Pull request by RLCASTRO**
**Describe the technical changes contained in this PR**
Updated the Meal.tsx page to query the ingredients from the database and display the data in the appropriate section

**Previous behaviour**
No Ingredients were displayed in the Meal.tsx page.

**New behaviour**
The Meal.tsx page displays the names of the ingredients, quantities, and units for each meal.
image

**Related issues addressed by this PR**
Fixes #609 

**Have the following been addressed?**
- [ ] Have test cases been created for all of the changes?
- [ ] Do all of the test cases pass?
- [ ] Has the testing been done using the default docker-compose environment?
- [ ] Are documentation changes required?
- [ ] Does this change break or alter existing behaviour?

### CODE REVIEW

**Comment by shanthisa** [comment]
Thanks for sharing the screenshot. This showed an important field we need to show. Need to show which is the substitute ingredient and which is the substitute it is for and what reason.
Eg:
The original recipe doc says Kosher or Regular salt 1 tsp
So we can show this as follows in our UI:

Kosher Salt 1 tsp
-------------------
Regular Salt 1 tsp
Substitute for: Kosher Salt 
Reason: not specified
--------------------

**Comment by RLCASTRO** [comment]
**Updated Image**

image

File: mealplanner-ui/src/pages/Meals/Meal.tsx [hunk 1]
```diff
@@ -273,14 +285,98 @@ export const Meal = () => {
...
             )}
             <Typography paddingBottom="1em">{meal?.descriptionFr}</Typography>
           </Grid>
           <Grid item xs={3}>
-            <Typography variant="h6"> Ingredients </Typography>
+            <Typography
+              variant="body1"
+              sx={{
+                "#ingredientsTable tbody tr:nth-of-type(even)": {
+                  backgroundColor: "#E8F5E9" /* Slightly darker green for even rows */,
```
Comment (shanthisa) [inline mealplanner-ui/src/pages/Meals/Meal.tsx:297]: use theme primary light color

File: mealplanner-ui/src/pages/Meals/Meal.tsx [hunk 2]
```diff
@@ -273,14 +285,98 @@ export const Meal = () => {
...
+                },
+                "#ingredientsTable": {
+                  border: "1px solid #E0E0E0",
```
Comment (shanthisa) [inline mealplanner-ui/src/pages/Meals/Meal.tsx:300]: use colors of theme

File: mealplanner-ui/src/pages/Meals/Meal.tsx [hunk 3]
```diff
@@ -273,14 +285,98 @@ export const Meal = () => {
...
+                  padding: "0.3em",
+                },
+                "#ingredientsTable thead": {
+                  borderBottom: "1px solid black",
```
Comment (shanthisa) [inline mealplanner-ui/src/pages/Meals/Meal.tsx:304]: refer theme

File: mealplanner-ui/src/pages/Meals/Meal.tsx [hunk 4]
```diff
@@ -273,14 +285,98 @@ export const Meal = () => {
...
+                "#ingredientsTable span": {
+                  fontStyle: "italic",
+
+                }
+              }}
+            >
+              <>
+                <table id="ingredientsTable" cellSpacing="0" cellPadding="0">
+                  <thead>
+                    <tr style={{ backgroundColor: "#E8F5E9" }}>
```
Comment (shanthisa) [inline mealplanner-ui/src/pages/Meals/Meal.tsx:318]: refer them

**Review by shanthisa — COMMENTED** [review]
Request the colour to be from the theme so that we know what colours we are using in an app level.

**Review by shanthisa — APPROVED** [review]
Approving this as it is only a cosmetic defect. Great work

**Pull request merged**